networkx/readwrite/tests/test_text.py

```diff
@@ -0,0 +1,263 @@
+import networkx as nx
+from textwrap import dedent
+
+
+def test_directed_tree_str():
+    # Create a directed forest with labels
+    graph = nx.balanced_tree(r=2, h=2, create_using=nx.DiGraph)
+    for node in graph.nodes:
+        graph.nodes[node]["label"] = "node_" + chr(ord("a") + node)
+
+    node_target = dedent(
+        """
+        ╙── 0
+            ├─╼ 1
+            │   ├─╼ 3
+            │   └─╼ 4
+            └─╼ 2
+                ├─╼ 5
+                └─╼ 6
+        """
+    ).strip()
+
+    label_target = dedent(
+        """
+        ╙── node_a
+            ├─╼ node_b
+            │   ├─╼ node_d
+            │   └─╼ node_e
+            └─╼ node_c
+                ├─╼ node_f
+                └─╼ node_g
+        """
+    ).strip()
+
+    # Basic node case
+    ret = nx.forest_str(graph, with_labels=False)
+    print(ret)
```

Comment:
Are these print statements intentionally put into the tests so when the tests fail you can actually see what is going on?  That's actually a nice idea. Is it true that pytest never actually shows these print results unless the test fails?

Reply:
The print statements serve several purposes. One is for debugging if the test fails. Another was for development. I needed to see the results (after manually verify they were correct of course) so I could populate the expected "want" fields with appropriately formatted text.

Pytest suppresses (doesn't show) output by default, but if you run with `-s` it will disable output suppression. So if you run the tests like this: 

```
pytest networkx/readwrite/tests/test_text.py -v -s
```

You can see the output even if the test does not fail.